ipc/provider/src/lib.rs

```diff
@@ -281,16 +282,24 @@ impl IpcProvider {
         subnet: SubnetID,
         from: Option<Address>,
         collateral: TokenAmount,
-        public_key: Vec<u8>,
     ) -> anyhow::Result<ChainEpoch> {
         let parent = subnet.parent().ok_or_else(|| anyhow!("no parent found"))?;
         let conn = self.get_connection(&parent)?;
 
         let subnet_config = conn.subnet();
         let sender = self.check_sender(subnet_config, from)?;
+        let addr = payload_to_evm_address(sender.payload())?;
+        let keystore = self.evm_wallet()?;
+        let key_info = keystore
+            .read()
+            .unwrap()
+            .get(&addr.into())?
+            .ok_or_else(|| anyhow!("key does not exists"))?;
+        let sk = libsecp256k1::SecretKey::parse_slice(key_info.private_key())?;
+        let public_key = libsecp256k1::PublicKey::from_secret_key(&sk).serialize();
 
```

Comment:
Since we don't input the public key any longer, can we at least print it out? That enable users to validate and/or copy the pubkey wherever they need it.

Reply:
Great idea I will add that feature